src/mono/wasm/runtime/roots.ts

```diff
@@ -287,8 +287,17 @@ class WasmJsOwnedRoot<T extends MonoObject> implements WasmRoot<T> {
     }
 
     set(value: T): T {
-        const destinationAddress = this.__buffer.get_address(this.__index);
-        cwraps.mono_wasm_write_managed_pointer_unsafe(destinationAddress, <ManagedPointer>value);
+        if (value === <any>0) {
+            // Don't use .set since it involves a gc write barrier, and the write barrier
```

Comment:
Are we concerned only about the case when the root is just newly allocated?
Or this is also when somebody free the root ?
Do you mean that it doesn't matter that GC on another thread would hold the root a bit longer ?
What is the semantics of `clear` method ?

Reply:
Clear zeroes out a root after you're done with it. This happens both when freeing the root (to avoid leaking object references) and when done with a reusable scratch root. The most common example of something where this would happen a lot is the root we use to store return values from function calls.